### PR TITLE
Simple wall turret buff

### LINKED ISSUE #32849: Turrets in walls are absolutely garbage at hitting anything.
If a turret is placed in a wall (as pretty much all shuttles typically do), it will almost always end up hitting the adjacent walls instead of the target that is literally one tile diagonal to it. Turrets typically have a slightly easier time hitting targets if they are placed within a horizontal line of walls (and obviously perform best when put in corners).

This may be considered balance, but it's not good when the turrets on the syndicate shuttle (or elsewhere) are physically incapable of hitting anything pretty much right next to them because hugging the walls makes them literally unable to hit you while you melee them to death.

### PR DESCRIPTION
**Pull request by AnturK**
Very simple and imprecise wall turret buff.

Closes #32849

### CODE REVIEW

**Comment by ShizCalev** [comment]
#32849?

**Comment by AnturK** [comment]
Yep

**Comment by Supermichael777** [comment]
this gives wall turrets 360 coverage which is hacky as fuck.  They should be properly one one side of the wall.

**Comment by AnturK** [comment]
It does, proper fix would require a lot of map edits.

**Comment by WJohn** [comment]
Then in future mapping we just need to put turrets on corners or otherwise unable to shoot inside if we don't want them to.

**Comment by AnturK** [comment]
Added mapper support for specific direction to shoot from, the previous functionality is the default when not set.

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -517,10 +533,13 @@
 
 
 	//Shooting Code:
-	A.preparePixelProjectile(target, src)
+	A.preparePixelProjectile(target, T)
```
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: does this not by itself fix it?
(I think dense objects can EXIT other dense objects...)

Comment (AnturK) [reply code/game/machinery/porta_turret/portable_turret.dm]: The issue is not with exiting the turf the turret is on but hitting the surrounding turfs on cardinal directions.

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**